EIPS/eip-6969.md

```diff
@@ -0,0 +1,169 @@
+---
+eip: 6969
+title: Contract Secured Revenue on an EVM based L2
+status: Draft
+type: Core
+discussions-to: https://t.me/+w1EKtAx824VlZDJh
+author: Zak Cole <zak@numbergroup.xyz>, Kevin Owocki <kevin@supermodular.xyz>
+created: 2023-05-01
+---
+
+## Abstract
+
+Contract Secured Revenue (CSR) allows smart contract developers to claim a percentage of all transaction fees paid by users when interacting with their smart contracts.
+
+This EIP proposes the introduction of CSR on EVM-based L2s. If adopted, this would allow smart contract developers who deploy on L2s to access a potentially lucrative revenue stream and/or fund public goods.
+
+## Motivation
+
+<!--
+  This section is optional.
+
+  The motivation section should include a description of any nontrivial problems the EIP solves. It should not describe how the EIP solves those problems, unless it is not immediately obvious. It should not describe why the EIP should be made into a standard, unless it is not immediately obvious.
+
+  With a few exceptions, external links are not allowed. If you feel that a particular resource would demonstrate a compelling case for your EIP, then save it as a printer-friendly PDF, put it in the assets folder, and link to that copy.
+
+  TODO: Remove this comment before submitting
+-->
+
+Using protocol rewards of an L1 to fund smart contract development would be a big change to the way the current market works.  This EIP *does not* advocate for any changes to the existing Ethereum L1.
+
+This EIP does advocate that L2s could begin to experiment with Contract Secured Revenue as a means of
+1. creating a new revenue stream for smart contract developers
+2. creating a new way of funding public goods
+3. creating incentives for developers to deploy their dapps on your network
+
+We acknowledge the potential downsides/risks of CSR.
+1. It could [become just a race to the bottom devolving into rebates for users](https://twitter.com/trent_vanepps/status/1623361559983452160)
+2. It has [yet to be proven that it can sustainable attract/retain/comp devs](https://twitter.com/trent_vanepps/status/1653086558633836564)
```

Comment:
You'll have to remove external links, except for the exceptions defined in EIP-1.